core/workspace_svg.js

```diff
@@ -1054,6 +1053,11 @@ Blockly.WorkspaceSvg.prototype.pasteBlock_ = function(xmlBlock) {
   Blockly.Events.disable();
   try {
     var block = Blockly.Xml.domToBlock(xmlBlock, this);
+    if (Blockly.Navigation.insertionConnection_) {
+      Blockly.Navigation.insertBlock(block, Blockly.Navigation.insertionConnection_);
+      return;
```

Comment:
Does the finally run if you return early inside a try block?  @NeilFraser

Reply:
Yes, it does.